### PR TITLE
Add browser-or-node types

### PR DESCRIPTION
**Pull request by jussikinnula**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [X] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by jussikinnula** [comment]
Note! There's already a PR (https://github.com/flexdinesh/browser-or-node/pull/6) for adding types into the package itself, but it's been open for a while. Let's add this and when the package gets types, then remove these.

File: types/browser-or-node/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Type definitions for browser-or-node 1.2
+// Project: https://github.com/flexdinesh/browser-or-node
+// Definitions by: Jussi Kinnula <https://github.com/jussikinnula>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.7
```
Comment (weswigham) [inline types/browser-or-node/index.d.ts]: You can omit this line from the header - these type definitions are super simple and should work all the way back into typescript's past.

**Review by weswigham — CHANGES_REQUESTED** [review]
The TS version header seems a bit gratuitous for these types

File: types/browser-or-node/index.d.ts (hunk 1 above)
Comment (jussikinnula) [reply types/browser-or-node/index.d.ts]: I changed the TS version to 2.8, as the tests were passing with that. You're right that it should work even with the oldest TypeScript versions.

**Review by weswigham — APPROVED** [review]

**Pull request merged**